### ISSUE TITLE
docker db-sync looses connection with node.socket, entering in a failing loop

### ISSUE THREAD

**Issue by stakelovelace** [input-output-hk/cardano-db-sync#984]
**OS**
Your OS:
Debian
**Versions**
The `db-sync` version (eg `cardano-db-sync --version`): 11.0.4
PostgreSQL version:
13.5
**Build/Install Method**
The method you use to build or install `cardano-db-sync`:
build
**Run method**
The method you used to run `cardano-db-sync` (eg Nix/Docker/systemd/none):
Docker
**Additional context**

**Problem Report**
Please do not include screenshots or images, but instead cut and paste any relevant log messages
or errors.

After some random time (from few hour to few days running) the db-sync container which gets connected to the docker cadano-node trough the "node.socket" file looses its connection and never recovers it entering in an infinite loop.

Here below the error message that keeps repeating.
```
[db-sync-node.Subscription:Warning:64] [2021-12-13 21:33:53.47 UTC] Identity Failed to start all required subscriptions
[db-sync-node.Subscription:Notice:64] [2021-12-13 21:33:54.48 UTC] Identity Restarting Subscription after 1.026744664s desired valency 1 current valency 0
[db-sync-node.Subscription:Notice:64] [2021-12-13 21:34:03.46 UTC] Identity Starting Subscription Worker, valency 1
[db-sync-node.Subscription:Notice:116697] [2021-12-13 21:34:03.46 UTC] Identity Connection Attempt Start, destination LocalAddress "/node-ipc/node.socket"
[db-sync-node.Subscription:Notice:116697] [2021-12-13 21:34:03.46 UTC] Identity Connection Attempt End, destination LocalAddress "/node-ipc/node.socket" outcome: ConnectSuccessLast
[db-sync-node.Handshake:Info:116697] [2021-12-13 21:34:03.46 UTC] WithMuxBearer (ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"}) Send (ClientAgency TokPropose,MsgProposeVersions (fromList [(NodeToClientV_1,TInt 764824073),(NodeToClientV_2,TInt 764824073),(NodeToClientV_3,TInt 764824073),(NodeToClientV_4,TInt 764824073),(NodeToClientV_5,TInt 764824073),(NodeToClientV_6,TInt 764824073),(NodeToClientV_7,TInt 764824073),(NodeToClientV_8,TInt 764824073),(NodeToClientV_9,TInt 764824073)]))
[db-sync-node.Handshake:Info:116697] [2021-12-13 21:34:03.46 UTC] WithMuxBearer (ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"}) Recv (ServerAgency TokConfirm,MsgAcceptVersion NodeToClientV_9 (TInt 764824073))
[db-sync-node.Mux:Info:116697] [2021-12-13 21:34:03.46 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"} event: Handshake Client end, duration 0.000596333s
[db-sync-node.Mux:Info:116698] [2021-12-13 21:34:03.46 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"} event: State: Mature
[db-sync-node:Info:116701] [2021-12-13 21:34:03.46 UTC] Starting chainSyncClient
[db-sync-node:Error:116701] [2021-12-13 21:34:03.46 UTC] ChainSyncWithBlocksPtcl: libpq: failed (no connection to the server
)
[db-sync-node.Mux:Info:116698] [2021-12-13 21:34:03.46 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"} event: State: Dead
[db-sync-node.Mux:Notice:116698] [2021-12-13 21:34:03.46 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"} event: Miniprotocol MiniProtocolNum 5 InitiatorDir terminated with exception libpq: failed (no connection to the server
)
[db-sync-node.Subscription:Error:116697] [2021-12-13 21:34:03.46 UTC] Identity Application Exception: LocalAddress "/node-ipc/node.socket" libpq: failed (no connection to the server
)
[db-sync-node.ErrorPolicy:Error:6] [2021-12-13 21:34:03.46 UTC] IP LocalAddress "/node-ipc/node.socket" ErrorPolicyUnhandledApplicationException libpq: failed (no connection to the server
)
[db-sync-node.Subscription:Warning:64] [2021-12-13 21:34:03.48 UTC] Identity Failed to start all required subscriptions
```

While here below you can see running the "ss -xa" command (within the official db-sync container) every second that the db-sync process is connecting to node.socket establishing and soon after dropping the connection.
This is an infinite loop that is not dependent from the cardano-node container which keeps running properly.
![photo_2021-12-13_13-23-20](https://user-images.githubusercontent.com/56354546/145821448-27a1f5d5-6fce-475b-aebf-9086fa301d2e.jpg)

This issue it is happening a lot in docker environment, hearing from other people too.
My system is a dedicated server Quad core i7 with 64Gb RAM.

**Comment by erikd** [comment]
Would be really interested to see the `node` logs for the same time.

**Comment by redoracle** [comment]
it just happened again attached the node's log file.

this catched my eyes, can be this related?:
{"thread":"1584","sev":"Error","data":{"address":"LocalAddress \"/opt/cardano/cnode/sockets/node.socket@2\"","kind":"ErrorPolicyTrace","event":"ErrorPolicyUnhandledApplicationException (MuxError MuxBearerClosed \"<socket: 29> closed when reading data, waiting on next header True\")"},"loc":null,"env":"1.31.0:2cbe3","msg":"","app":[],"host":"5490e33f","at":"2021-12-15T19:22:43.16Z","ns":["cardano.node.LocalErrorPolicy"],"pid":"279"

[node0-20211215162450.json.zip](https://github.com/input-output-hk/cardano-db-sync/files/7723579/node0-20211215162450.json.zip)

**Comment by DamjanOstrelic** [comment]
Same issue here - cardano-node `1.31.0`, db-sync: `12.0.0`

db-sync logs:
```
cardano-db-sync_1  | [db-sync-node.Handshake:Info:3776] [2021-12-26 11:02:23.35 UTC] WithMuxBearer (ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"}) Send (ClientAgency TokPropose,MsgProposeVersions (fromList [(NodeToClientV_1,TInt 764824073),(NodeToClientV_2,TInt 764824073),(NodeToClientV_3,TInt 764824073),(NodeToClientV_4,TInt 764824073),(NodeToClientV_5,TInt 764824073),(NodeToClientV_6,TInt 764824073),(NodeToClientV_7,TInt 764824073),(NodeToClientV_8,TInt 764824073),(NodeToClientV_9,TInt 764824073),(NodeToClientV_10,TInt 764824073)]))
cardano-db-sync_1  | [db-sync-node.Handshake:Info:3776] [2021-12-26 11:02:23.35 UTC] WithMuxBearer (ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"}) Recv (ServerAgency TokConfirm,MsgAcceptVersion NodeToClientV_10 (TInt 764824073))
cardano-db-sync_1  | [db-sync-node.Mux:Info:3776] [2021-12-26 11:02:23.35 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"} event: Handshake Client end, duration 0.00107852s
cardano-db-sync_1  | [db-sync-node.Mux:Info:3777] [2021-12-26 11:02:23.36 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"} event: State: Mature
cardano-db-sync_1  | [db-sync-node:Info:3780] [2021-12-26 11:02:23.36 UTC] Starting chainSyncClient
cardano-db-sync_1  | [db-sync-node:Error:3780] [2021-12-26 11:02:23.36 UTC] ChainSyncWithBlocksPtcl: libpq: failed (no connection to the server
cardano-db-sync_1  | )
cardano-db-sync_1  | [db-sync-node.Mux:Info:3777] [2021-12-26 11:02:23.36 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"} event: State: Dead
cardano-db-sync_1  | [db-sync-node.Mux:Notice:3777] [2021-12-26 11:02:23.36 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"} event: Miniprotocol MiniProtocolNum 5 InitiatorDir terminated with exception libpq: failed (no connection to the server
cardano-db-sync_1  | )
cardano-db-sync_1  | [db-sync-node.Subscription:Error:3776] [2021-12-26 11:02:23.36 UTC] Identity Application Exception: LocalAddress "/node-ipc/node.socket" libpq: failed (no connection to the server
cardano-db-sync_1  | )
cardano-db-sync_1  | [db-sync-node.ErrorPolicy:Error:6] [2021-12-26 11:02:23.36 UTC] IP LocalAddress "/node-ipc/node.socket" ErrorPolicyUnhandledApplicationException libpq: failed (no connection to the server
cardano-db-sync_1  | )
cardano-db-sync_1  | [db-sync-node.Subscription:Warning:50] [2021-12-26 11:02:23.38 UTC] Identity Failed to start all required subscriptions
cardano-db-sync_1  | [db-sync-node.Subscription:Notice:50] [2021-12-26 11:02:24.38 UTC] Identity Restarting Subscription after 1.026531629s desired valency 1 current valency 0
```
cardano-node logs:
```
{"thread":"288","sev":"Error","data":{"address":"LocalAddress \"/opt/cardano/cnode/sockets/node.socket@283\"","kind":"ErrorPolicyTrace","event":"ErrorPolicyUnhandledApplicationException (MuxError MuxBearerClosed \"<socket: 30> closed when reading data, waiting on next header True\")"},"loc":null,"env":"1.31.0:2cbe3","msg":"","app":[],"host":"b19c20f0","at":"2021-12-26T11:01:14.31Z","ns":["cardano.node.LocalErrorPolicy"],"pid":"4579"}
{"thread":"288","sev":"Error","data":{"address":"LocalAddress \"/opt/cardano/cnode/sockets/node.socket@284\"","kind":"ErrorPolicyTrace","event":"ErrorPolicyUnhandledApplicationException (MuxError MuxBearerClosed \"<socket: 40> closed when reading data, waiting on next header True\")"},"loc":null,"env":"1.31.0:2cbe3","msg":"","app":[],"host":"b19c20f0","at":"2021-12-26T11:01:24.33Z","ns":["cardano.node.LocalErrorPolicy"],"pid":"4579"}
{"thread":"288","sev":"Error","data":{"address":"LocalAddress \"/opt/cardano/cnode/sockets/node.socket@285\"","kind":"ErrorPolicyTrace","event":"ErrorPolicyUnhandledApplicationException (MuxError MuxBearerClosed \"<socket: 50> closed when reading data, waiting on next header True\")"},"loc":null,"env":"1.31.0:2cbe3","msg":"","app":[],"host":"b19c20f0","at":"2021-12-26T11:01:34.32Z","ns":["cardano.node.LocalErrorPolicy"],"pid":"4579"}
```

**Comment by thakkery** [comment]
I'm also having this issue (though it took me a while to realize it was this issue). More logs below:

```
cardano-db-sync_1  | [db-sync-node.Subscription:Notice:46] [2022-01-03 04:19:35.28 UTC] Identity Starting Subscription Worker, valency 1
cardano-db-sync_1  | [db-sync-node.Subscription:Notice:59] [2022-01-03 04:19:35.28 UTC] Identity Connection Attempt Start, destination LocalAddress "/node-ipc/node.socket"
cardano-db-sync_1  | [db-sync-node.Subscription:Error:59] [2022-01-03 04:19:35.28 UTC] Identity Connection Attempt Exception, destination LocalAddress "/node-ipc/node.socket" exception: Network.Socket.connect: <socket: 22>: does not exist (No such file or directory)
cardano-db-sync_1  | [db-sync-node.ErrorPolicy:Warning:6] [2022-01-03 04:19:35.28 UTC] IP LocalAddress "/node-ipc/node.socket" ErrorPolicySuspendPeer (Just (ConnectionExceptionTrace Network.Socket.connect: <socket: 22>: does not exist (No such file or directory))) 20s 20s
cardano-db-sync_1  | [db-sync-node.Subscription:Warning:46] [2022-01-03 04:19:35.31 UTC] Identity Failed to start all required subscriptions
cardano-db-sync_1  | [db-sync-node.Subscription:Notice:46] [2022-01-03 04:19:36.31 UTC] Identity Restarting Subscription after 1.02666558s desired valency 1 current valency 0
cardano-db-sync_1  | [db-sync-node.Subscription:Notice:46] [2022-01-03 04:19:45.29 UTC] Identity Starting Subscription Worker, valency 1
cardano-db-sync_1  | [db-sync-node.Subscription:Notice:61] [2022-01-03 04:19:45.29 UTC] Identity Connection Attempt Start, destination LocalAddress "/node-ipc/node.socket"
cardano-db-sync_1  | [db-sync-node.Subscription:Error:61] [2022-01-03 04:19:45.29 UTC] Identity Connection Attempt Exception, destination LocalAddress "/node-ipc/node.socket" exception: Network.Socket.connect: <socket: 22>: does not exist (No such file or directory)
cardano-db-sync_1  | [db-sync-node.ErrorPolicy:Warning:6] [2022-01-03 04:19:45.29 UTC] IP LocalAddress "/node-ipc/node.socket" ErrorPolicySuspendPeer (Just (ConnectionExceptionTrace Network.Socket.connect: <socket: 22>: does not exist (No such file or directory))) 20s 20s
cardano-db-sync_1  | [db-sync-node.Subscription:Warning:46] [2022-01-03 04:19:45.32 UTC] Identity Failed to start all required subscriptions

```

**Comment by rcmorano** [comment]
> Same issue here - cardano-node `1.31.0`, db-sync: `12.0.0`
> 
> db-sync logs:
> 
> ```
> cardano-db-sync_1  | [db-sync-node.Handshake:Info:3776] [2021-12-26 11:02:23.35 UTC] WithMuxBearer (ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"}) Send (ClientAgency TokPropose,MsgProposeVersions (fromList [(NodeToClientV_1,TInt 764824073),(NodeToClientV_2,TInt 764824073),(NodeToClientV_3,TInt 764824073),(NodeToClientV_4,TInt 764824073),(NodeToClientV_5,TInt 764824073),(NodeToClientV_6,TInt 764824073),(NodeToClientV_7,TInt 764824073),(NodeToClientV_8,TInt 764824073),(NodeToClientV_9,TInt 764824073),(NodeToClientV_10,TInt 764824073)]))
> cardano-db-sync_1  | [db-sync-node.Handshake:Info:3776] [2021-12-26 11:02:23.35 UTC] WithMuxBearer (ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"}) Recv (ServerAgency TokConfirm,MsgAcceptVersion NodeToClientV_10 (TInt 764824073))
> cardano-db-sync_1  | [db-sync-node.Mux:Info:3776] [2021-12-26 11:02:23.35 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"} event: Handshake Client end, duration 0.00107852s
> cardano-db-sync_1  | [db-sync-node.Mux:Info:3777] [2021-12-26 11:02:23.36 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"} event: State: Mature
> cardano-db-sync_1  | [db-sync-node:Info:3780] [2021-12-26 11:02:23.36 UTC] Starting chainSyncClient
> cardano-db-sync_1  | [db-sync-node:Error:3780] [2021-12-26 11:02:23.36 UTC] ChainSyncWithBlocksPtcl: libpq: failed (no connection to the server
> cardano-db-sync_1  | )
> cardano-db-sync_1  | [db-sync-node.Mux:Info:3777] [2021-12-26 11:02:23.36 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"} event: State: Dead
> cardano-db-sync_1  | [db-sync-node.Mux:Notice:3777] [2021-12-26 11:02:23.36 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"} event: Miniprotocol MiniProtocolNum 5 InitiatorDir terminated with exception libpq: failed (no connection to the server
> cardano-db-sync_1  | )
> cardano-db-sync_1  | [db-sync-node.Subscription:Error:3776] [2021-12-26 11:02:23.36 UTC] Identity Application Exception: LocalAddress "/node-ipc/node.socket" libpq: failed (no connection to the server
> cardano-db-sync_1  | )
> cardano-db-sync_1  | [db-sync-node.ErrorPolicy:Error:6] [2021-12-26 11:02:23.36 UTC] IP LocalAddress "/node-ipc/node.socket" ErrorPolicyUnhandledApplicationException libpq: failed (no connection to the server
> cardano-db-sync_1  | )
> cardano-db-sync_1  | [db-sync-node.Subscription:Warning:50] [2021-12-26 11:02:23.38 UTC] Identity Failed to start all required subscriptions
> cardano-db-sync_1  | [db-sync-node.Subscription:Notice:50] [2021-12-26 11:02:24.38 UTC] Identity Restarting Subscription after 1.026531629s desired valency 1 current valency 0
> ```
> 
> cardano-node logs:
> 
> ```
> {"thread":"288","sev":"Error","data":{"address":"LocalAddress \"/opt/cardano/cnode/sockets/node.socket@283\"","kind":"ErrorPolicyTrace","event":"ErrorPolicyUnhandledApplicationException (MuxError MuxBearerClosed \"<socket: 30> closed when reading data, waiting on next header True\")"},"loc":null,"env":"1.31.0:2cbe3","msg":"","app":[],"host":"b19c20f0","at":"2021-12-26T11:01:14.31Z","ns":["cardano.node.LocalErrorPolicy"],"pid":"4579"}
> {"thread":"288","sev":"Error","data":{"address":"LocalAddress \"/opt/cardano/cnode/sockets/node.socket@284\"","kind":"ErrorPolicyTrace","event":"ErrorPolicyUnhandledApplicationException (MuxError MuxBearerClosed \"<socket: 40> closed when reading data, waiting on next header True\")"},"loc":null,"env":"1.31.0:2cbe3","msg":"","app":[],"host":"b19c20f0","at":"2021-12-26T11:01:24.33Z","ns":["cardano.node.LocalErrorPolicy"],"pid":"4579"}
> {"thread":"288","sev":"Error","data":{"address":"LocalAddress \"/opt/cardano/cnode/sockets/node.socket@285\"","kind":"ErrorPolicyTrace","event":"ErrorPolicyUnhandledApplicationException (MuxError MuxBearerClosed \"<socket: 50> closed when reading data, waiting on next header True\")"},"loc":null,"env":"1.31.0:2cbe3","msg":"","app":[],"host":"b19c20f0","at":"2021-12-26T11:01:34.32Z","ns":["cardano.node.LocalErrorPolicy"],"pid":"4579"}
> ```

Note the `libpq` (postgres lib) error here. The node socket error is thrown after some exception is raised from the database thread. At least that's something that happens to me (check [0]) when my postgres is restarted and db-sync loses connectivity to it. It never recovers until I restart db-sync and it has always thrown these misleading (or maybe real, but produced cos of db error somehow) node socket errors.

[0]
```
[db-sync-node.Mux:Info:67] [2021-12-29 15:02:21.51 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/ipc/node.socket"} event: State: Mature
[db-sync-node:Info:70] [2021-12-29 15:02:21.51 UTC] Starting chainSyncClient
[db-sync-node.Subscription:Notice:65] [2021-12-29 15:02:21.54 UTC] Identity Required subscriptions started
[db-sync-node:Info:70] [2021-12-29 15:02:21.54 UTC] Cardano.Db tip is at slot 49209392, block 6689128
[db-sync-node:Info:75] [2021-12-29 15:02:21.54 UTC] Running DB thread
[db-sync-node:Info:75] [2021-12-29 15:02:21.57 UTC] Rolling back to slot 48988785, hash 53946e7ac007cbe09c7189ca143cc6988f0bcab566550fb309796b104f464e88
[db-sync-node:Info:75] [2021-12-29 15:02:21.69 UTC] Deleting 10683 blocks up to slot 49209392
[db-sync-node:Error:75] [2021-12-29 22:43:57.49 UTC] runDBThread: libpq: failed (no connection to the server
)
[db-sync-node:Error:78] [2021-12-29 22:43:57.49 UTC] recvMsgRollBackward: AsyncCancelled
[db-sync-node:Error:70] [2021-12-29 22:43:57.49 UTC] ChainSyncWithBlocksPtcl: libpq: failed (no connection to the server
)
[db-sync-node.Mux:Info:67] [2021-12-29 22:43:57.49 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/ipc/node.socket"} event: State: Dead
[db-sync-node.Mux:Notice:67] [2021-12-29 22:43:57.49 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/ipc/node.socket"} event: Miniprotocol MiniProtocolNum 5 InitiatorDir terminated with exception libpq: failed (no connection to the server
)
[db-sync-node.Subscription:Error:66] [2021-12-29 22:43:57.49 UTC] Identity Application Exception: LocalAddress "/node-ipc/node.socket" libpq: failed (no connection to the server
)
[db-sync-node.ErrorPolicy:Error:6] [2021-12-29 22:43:57.49 UTC] IP LocalAddress "/node-ipc/node.socket" ErrorPolicyUnhandledApplicationException libpq: failed (no connection to the server
```

**Comment by thakkery** [comment]
This does happen usually after I restart my machine. Does the docker-compose file not restart everything on the same network? Maybe it would help if there were explicit port bindings to the node socket in the `docker-compose.yml` file

**Comment by nakedpony** [comment]
I'm also having this issue. I'm using docker-compose and `inputoutput/cardano-db-sync:12.0.1` with `inputoutput/cardano-node:1.31.0`. Tried restarting everything but no luck.
Is there any workaround?

**Comment by andreabravetti** [comment]
Today I had the same problem on a testnet node 1.33.0 with a db-sync attached.

cardano-node log (this line repeated...):

`
Feb 22 10:52:23 gandalf.goada.net cardano-node[788]: [gandalf:cardano.node.LocalErrorPolicy:Error:278] [2022-02-22 09:52:23.25 UTC] IP LocalAddress "/home/cardano/testnet/node/db/node.socket@9373" ErrorPolicyUnhandledApplicationException (MuxError MuxBearerClosed "<socket: 50> closed >
`

db-sync log:

`
Feb 22 11:01:32 gandalf.goada.net cardano-db-sync[1505]: [db-sync-node.Subscription:Notice:49] [2022-02-22 10:01:32.69 UTC] Identity Starting Subscription Worker, valency 1
Feb 22 11:01:32 gandalf.goada.net cardano-db-sync[1505]: [db-sync-node.Subscription:Notice:253647] [2022-02-22 10:01:32.69 UTC] Identity Connection Attempt Start, destination LocalAddress "/home/cardano/testnet/node/db/node.socket"
Feb 22 11:01:32 gandalf.goada.net cardano-db-sync[1505]: [db-sync-node.Subscription:Error:253647] [2022-02-22 10:01:32.69 UTC] Identity Connection Attempt Exception, destination LocalAddress "/home/cardano/testnet/node/db/node.socket" exception: Network.Socket.connect: <socket: 21>: d>
Feb 22 11:01:32 gandalf.goada.net cardano-db-sync[1505]: [db-sync-node.ErrorPolicy:Warning:6] [2022-02-22 10:01:32.69 UTC] IP LocalAddress "/home/cardano/testnet/node/db/node.socket" ErrorPolicySuspendPeer (Just (ConnectionExceptionTrace Network.Socket.connect: <socket: 21>: does not >
Feb 22 11:01:32 gandalf.goada.net cardano-db-sync[1505]: [db-sync-node.Subscription:Warning:49] [2022-02-22 10:01:32.72 UTC] Identity Failed to start all required subscriptions
`

**Comment by ssledz** [comment]
I have the same problem either on testnet and mainnet. 

I am using following images:
- inputoutput/cardano-db-sync:12.0.2
- inputoutput/cardano-node:1.33.1

**Comment by tmcgilchrist** [comment]
Looking at this in more detail what is happening for me is:
  
   1. cardano-db-sync starts up before Postgres is ready and attempts to run [getLatestPoints](https://github.com/input-output-hk/cardano-db-sync/blob/master/cardano-db-sync/src/Cardano/DbSync/Sync.hs#L225) which errors out with
      `ChainSyncWithBlocksPtcl: libpq: failed (no connection to the server)`

   ``` shell
[db-sync-node:Info:2074068] [2022-03-05 06:22:14.59 UTC] Starting chainSyncClient
[db-sync-node:Error:2074068] [2022-03-05 06:22:14.59 UTC] ChainSyncWithBlocksPtcl: libpq: failed (no connection to the server)
[db-sync-node.Mux:Info:2074065] [2022-03-05 06:22:14.59 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "state-node-mainnet/node.socket"} event: State: Dead
[db-sync-node.Mux:Notice:2074065] [2022-03-05 06:22:14.59 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "state-node-mainnet/node.socket"} event: Miniprotocol MiniProtocolNum 5 InitiatorDir terminated with exception libpq: failed (no connection to the server)
[db-sync-node.Subscription:Error:2074064] [2022-03-05 06:22:14.59 UTC] Identity Application Exception: LocalAddress "../cardano-node/state-node-mainnet/node.socket" libpq: failed (no connection to the server)
[db-sync-node.ErrorPolicy:Error:6] [2022-03-05 06:22:14.59 UTC] IP LocalAddress "../cardano-node/state-node-mainnet/node.socket" ErrorPolicyUnhandledApplicationException libpq: failed (no connection to the server)
[db-sync-node.Subscription:Warning:54] [2022-03-05 06:22:14.61 UTC] Identity Failed to start all required subscriptions
[db-sync-node.Subscription:Notice:54] [2022-03-05 06:22:15.61 UTC] Identity Restarting Subscription after 1.027235s desired valency 1 current valency 0   
   ```

   2. Due to that error the IPC connection to cardano-node doesn't get established and `cardano-node` sees an `ErrorPolicyUnhandledApplicationException` 

   ``` shell
[noir:cardano.node.LocalHandshake:Info:526264] [2022-03-05 06:22:14.59 UTC] WithMuxBearer (ConnectionId {localAddress = LocalAddress "state-node-mainnet/node.socket", remoteAddress = LocalAddress "state-node-mainnet/node.socket@62181"}) Send (ServerAgency TokConfirm,MsgAcceptVersion NodeToClientV_12 (TInt 764824073))
[noir:cardano.node.LocalErrorPolicy:Error:65] [2022-03-05 06:22:15.59 UTC] IP LocalAddress "state-node-mainnet/node.socket@62181" ErrorPolicyUnhandledApplicationException (MuxError MuxBearerClosed "<socket: 27> closed when reading data, waiting on next header True")   
   ```
   
   3. cardano-db-sync enters a loop retrying the psql connection and cardano-node connection but never succeeding. In my case it was an issue with Postgres running out of disk space.
      
Work around for now would be adding healthcheck sections to docker-compose file.
See https://github.com/input-output-hk/cardano-db-sync/pull/1077 for the suggested changes to make.

**Comment by erikd** [comment]
This is likely to be fixed by commit fbae4ad78c204d10201cdc43da98987a554a10fd that will be in the next release.
